tsconfig.json

```diff
@@ -16,5 +16,6 @@
 		"sourceMap": true,
 		"declarationMap": true,
 		"strict": true,
-  },
-}
\ No newline at end of file
+		"esModuleInterop": true
```

Comment:
Enabling this also enables [allowSyntheticDefaultImports](https://www.typescriptlang.org/tsconfig#allowSyntheticDefaultImports). We need to set it `esModuleInterop` to `false` which is the default value for the compiler. Importing a module with custom name as default import is a not a good practice, end up having different names for same package on different code files.

Reply:
I enabled this for allowing default import of:
`../../node_modules/ethereumjs-util/dist/constants.d.ts(2,8): error TS1259: Module '"/web3.js/node_modules/@types/bn.js/index"' can only be default-imported using the 'esModuleInterop' flag`

Reply:
I could not find the source of this using the default import. 

https://github.com/ethereumjs/ethereumjs-monorepo/blob/0c7bdfcdf5866fc31df2b87508156f3294bbae45/packages/util/src/constants.ts#L1-L2

If you are using it somewhere, you can import such with syntax as `import * as name from 'package'` without enabling this option.

Reply:
its not used in our package but as error is indicating its in node_modules, could you help me and check this branch on your system with esModuleInterop to false and see its reported by lib in node_modules

Reply:
I fixed it by changing the `tsconfig.json` only for that package where it cause error. 

Reported and issue to etherumjs  https://github.com/ethereumjs/ethereumjs-monorepo/issues/1615 and also created an issue on our repo for tracking https://github.com/ChainSafe/web3.js/issues/4636